core/src/transport.rs

```diff
@@ -149,9 +149,22 @@ pub trait Transport {
         cx: &mut Context<'_>,
     ) -> Poll<TransportEvent<Self::ListenerUpgrade, Self::Error>>;
 
-    /// Performs a transport-specific mapping of an address `observed` by
-    /// a remote onto a local `listen` address to yield an address for
-    /// the local node that may be reachable for other peers.
+    /// Performs a transport-specific mapping of an address `observed` by a remote onto a
+    /// local `listen` address to yield an address for the local node that may be reachable
+    /// for other peers.
+    ///
+    /// This is relevant for transports where Network Address Translation (NAT) can occur
+    /// so that e.g. the peer is observed at a different IP than the IP of the local
+    /// listening address. See also [`address_translation`][crate::address_translation].
+    ///
+    /// Within [`libp2p::Swarm`](<https://docs.rs/libp2p/latest/libp2p/struct.Swarm.html>) this is
```

Comment:
Is there a better way to link to items from crates that are not a dependency?

Reply:
I am not aware of a way. Maybe @thomaseizinger?

Reply:
Me neither but I also think it is kind of an odd thing to do. This crate shouldn't "know" about `Swarm`. Do we have to include this sentence here?

Reply:
> Me neither but I also think it is kind of an odd thing to do. This crate shouldn't "know" about `Swarm`. Do we have to include this sentence here?

Good point. But in practice transports are commonly used in a swarm, so I think its okay to mention this here. When I was looking at it for Quic it helped me to know how it is used in the swarm.